tests/functional/adapter/test_basic.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+
+from dbt.tests.adapter.basic.test_base import BaseSimpleMaterializations
+from dbt.tests.adapter.basic.test_singular_tests import BaseSingularTests
+from dbt.tests.adapter.basic.test_singular_tests_ephemeral import (
+    BaseSingularTestsEphemeral,
+)
+from dbt.tests.adapter.basic.test_empty import BaseEmpty
+from dbt.tests.adapter.basic.test_ephemeral import BaseEphemeral
+from dbt.tests.adapter.basic.test_incremental import BaseIncremental
+from dbt.tests.adapter.basic.test_generic_tests import BaseGenericTests
+from dbt.tests.adapter.basic.test_snapshot_check_cols import BaseSnapshotCheckCols
+from dbt.tests.adapter.basic.test_snapshot_timestamp import BaseSnapshotTimestamp
+
+from dbt.tests.adapter.basic.files import seeds_base_csv, seeds_added_csv, seeds_newcolumns_csv
+
+# set the datatype of the name column in the 'added' seed so it
+# can hold the '_update' that's added
+schema_seed_added_yml = """
+version: 2
+seeds:
+  - name: added
+    config:
+      column_types:
+        name: varchar(64)
+"""
+
+
+class TestSimpleMaterializationsRedshift(BaseSimpleMaterializations):
+    pass
+
+
+class TestSingularTestsRedshift(BaseSingularTests):
+    pass
+
+
+class TestSingularTestsEphemeralRedshift(BaseSingularTestsEphemeral):
+    pass
+
+
+class TestEmptyRedshift(BaseEmpty):
+    pass
+
+
+class TestEphemeralRedshift(BaseEphemeral):
+    pass
+
+
+class TestIncrementalRedshift(BaseIncremental):
+    pass
+
+
+class TestGenericTestsRedshift(BaseGenericTests):
+    pass
+
+
+class TestSnapshotCheckColsRedshift(BaseSnapshotCheckCols):
+    # Redshift defines the 'name' column such that it's not big enough
```

Comment:
Should we consider updating the base case to accommodate all adapters? but can also leave this as future steps

Reply:
We can't update the base case to make Redshift work, because it requires a specific column datatype which doesn't work for all adapters.

Reply:
I do think this is related to [some jank around updating](https://github.com/dbt-labs/dbt-core/blob/ea5a9da71ef71cc13414cd8c2f8097448add5edb/core/dbt/tests/util.py#L261-L327) these fixture tables mid-sequence, which is worth discussing further, alongside the jank we've already identified: https://github.com/dbt-labs/dbt-core/issues/4778#issuecomment-1075515724